### ISSUE TITLE
Add supported features to Pulsar Feature Matrix 

### ISSUE THREAD

**Issue by Anonymitaet** [fsharplang-ru/pulsar-client-dotnet#221]
Hi @Lanayx 

Thanks for making contributions to [pulsar-client-dotnet](https://github.com/fsprojects/pulsar-client-dotnet)!

To show features supported by [pulsar-client-dotnet](https://github.com/fsprojects/pulsar-client-dotnet) to users more clearly, we (+@RobertIndie) have added both `code` and `doc` info to the [Pulsar Feature Matrix](https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914&range=T:U) based on this [README](https://github.com/fsprojects/pulsar-client-dotnet#readme).

🔷 🔷 🔷

To ensure that the info is accurate, can you please：

1. Double check the info in the [`code`](https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914&range=T:T) column?

2. Double check: currently, we do have docs for these supported features?

🔷 🔷 🔷

Feel free to tick the box if we support more functions in the future. Thank you!

**Comment by Lanayx** [comment]
Hi! The table is pretty accurate, the only thing that I've found is User-defined Properties, that [are supported](https://github.com/fsprojects/pulsar-client-dotnet/blob/d643e7a2518b7ab6bd5a346c1836d944c3b557f8/src/Pulsar.Client/Api/IProducer.fs#L57) for Consumer, Producer and Reader
As for documentation - it was a deliberate decision to keep the same API as Java client so I didn't have to duplicate it.

**Comment by RobertIndie** [comment]
Hi, @Lanayx Thanks for your check!

> the only thing that I've found is User-defined Properties, that [are supported](https://github.com/fsprojects/pulsar-client-dotnet/blob/d643e7a2518b7ab6bd5a346c1836d944c3b557f8/src/Pulsar.Client/Api/IProducer.fs#L57) for Consumer, Producer and Reader

It seems only the [properties of the message](https://github.com/apache/pulsar/blob/5d6932137d76d544f939bef27df25f61b4a4d00d/pulsar-common/src/main/proto/PulsarApi.proto#L109) are supported. But setting the metadata of the [producer](https://github.com/apache/pulsar/blob/5d6932137d76d544f939bef27df25f61b4a4d00d/pulsar-common/src/main/proto/PulsarApi.proto#L479), [consumer and reader](https://github.com/apache/pulsar/blob/5d6932137d76d544f939bef27df25f61b4a4d00d/pulsar-common/src/main/proto/PulsarApi.proto#L363) seems not yet supported.

**Comment by Lanayx** [comment]
Ah, sorry, I wasn't aware of this feature. Then everything is correct, thank you!

**Comment by Anonymitaet** [comment]
@Lanayx thanks for your confirmation. Even though .net client is designed similar to Java client, it still deserves independent docs to instruct users on why/how to use it to alleviate users' concerns.

**Comment by Lanayx** [comment]
Yes, ideally it should have an independent doc and a person who adds it keeps it up to date. So if someone makes this effort, I'll endorse it. But from my experience - current situation is pretty ok - there were no configuration questions for the library since it's inception except of client logging (which is ironically missing on Java side as well)
